src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7460,6 +7460,22 @@ def make_nunique(identifier: str, dropna: bool) -> SnowparkColumn:
         )
         return SnowflakeQueryCompiler(frame)
 
+    def nunique_index(self, dropna: bool) -> "SnowflakeQueryCompiler":
+        """
+        Return number of unique elements in an Index object.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler where the first row in the first column contains the number
+        of unique elements.
+        """
+        return SnowflakeQueryCompiler(
```

Comment:
that is better, but if you are already at the query compiler layer, why not extend the _nunique_columns(dropna) function to handle the index columns, and i don't think you need any transpose here also, there seems no need to append any new columns there